template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/core/CoreConfigProvider.java

```diff
@@ -137,9 +143,7 @@ public List<String> getRoleTypes() {
 
     @Override
     public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
-        return !cmTemplateProcessor.isRoleTypePresentInService(HDFS, Lists.newArrayList(NAMENODE))
-                && source.getFileSystemConfigurationView().isPresent()
-                && ConfigUtils.getStorageLocationForServiceProperty(source, CORE_DEFAULTFS).isPresent();
+        return true;
```

Comment:
Why move this condition in to two different parts? This should be the only part where we logically apply a config change. If you change a different config value it should be a different ConfigProvider.